Source/CarthageKit/BuildSettings.swift

```diff
@@ -152,18 +170,48 @@ public struct BuildSettings {
 		}
 	}
 
+	private var productsDirectoryURLDependingOnAction: Result<URL, CarthageError> {
+		if action == .archive {
+			return self["OBJROOT"]
+				.fanout(archiveIntermediatesBuildProductsPath)
+				.map { objroot, path -> URL in
+					let root = URL(fileURLWithPath: objroot, isDirectory: true)
+					return root.appendingPathComponent(path)
+				}
+		} else {
+			return builtProductsDirectoryURL
+		}
+	}
+
+	private var archiveIntermediatesBuildProductsPath: Result<String, CarthageError> {
+		let r1 = self["TARGET_NAME"]
+		guard let schemeOrTarget = arguments.scheme ?? r1.value else { return r1 }
+
+		let r2 = self["CONFIGURATION"]
+		guard let configuration = r2.value else { return r2 }
+
+		let effectivePlatformName = self["EFFECTIVE_PLATFORM_NAME"].value ?? ""
```

Comment:
Could we add a comment to make explicit the potential lack of append?

```
/// a value almost certainly beginning with `-` or (lacking said value) an 
/// empty string to append without effect in the path below because Xcode
/// expects the path like that.
```

Otherwise, people might question: “why not a force-unwrap?”